test/Microsoft.AspNetCore.Razor.Tools.Test/Roslyn/BuildProtocolTest.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CommandLine;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Tools
+{
+    public class BuildProtocolTest
```

Comment:
What's BuildProtocol?

Reply:
The mega class that contains `BuildRequest`, `BuildResponse` and their implementations.

Reply:
ewww mega class.

If you're introducing tests for it now, why not also include the split/cleanup. You'll just have more to cleanup this way.

Reply:
Sure yea that makes sense 👍